app/scripts/modules/netflix/availability/availability.directive.ts

```diff
@@ -0,0 +1,108 @@
+
+import './availability.less';
+
+interface IAggregateDetails {
+  score: number;
+  reason: string;
+}
+
+export class AvailabilityController implements ng.IComponentController {
+  private activeRefresher: any;
+
+  static get $inject() {
+    return ['$scope', 'availabilityReaderService', 'schedulerFactory'];
+  }
+
+  public constructor (private $scope: any, private availabilityReaderService: AvailabilityReaderService, private schedulerFactory: any) {}
+
+  private getWindowScore (window: IAvailabilityWindow): number {
+    if (window.nines >= window.target_nines) { return 1; }
+    if (window.nines >= window.target_nines * 0.95) { return 2; }
+    return 4;
+  }
+
+  private getAggregateScore (result: IAvailabilityData): IAggregateDetails {
+    let score = 1;
+    let reason = '91 DAY GOAL MET!';
+
+    // Figure out score
+    const currentOrRecentIncidents = false; // TODO: Support handling current or recent incidents
+    if (currentOrRecentIncidents) {
+      // TODO: Add Current or recent incident links to details
+    } else if (result['yesterday'].score > 1) {
+      // If there were recent incidents yesterday
+      score = 4;
+      reason = '24 HOUR GOAL <strong>NOT</strong> MET';
+      // TODO: Add incident links to details
+    } else if (result['28days'].score > 1) {
+      // If we have not acheived 28 day availability goals
+      score = 3;
+      reason = '28 DAY GOAL <strong>NOT</strong> MET';
+    } else if (result['91days'].score > 1) {
+      // If we have not acheived 90 day availability goals (but rest are acheived)
+      score = 2;
+      reason = '91 DAY GOAL <strong>NOT</strong> MET';
+    }
+
+    return { score, reason };
+  }
+
+  public refreshData(): void {
+    this.availabilityReaderService.getAvailabilityData().then((result) => {
+      if (result) {
+        // Build composite availability score for main button
+        this.$scope.lastUpdated = result.last_updated;
+        this.$scope.yesterday = result.yesterday;
+        this.$scope.yesterday.score = this.getWindowScore(this.$scope.yesterday);
+        this.$scope.twentyeightdays = result['28days'];
+        this.$scope.twentyeightdays.score = this.getWindowScore(this.$scope.twentyeightdays);
+        this.$scope.ninetyonedays = result['91days'];
+        this.$scope.ninetyonedays.score = this.getWindowScore(this.$scope.ninetyonedays);
+
+        const aggregate: IAggregateDetails = this.getAggregateScore(result);
+        this.$scope.aggregate = aggregate;
+      }
+    });
+  }
+
+  public initialize(): void {
+    this.activeRefresher = this.schedulerFactory.createScheduler(30000);
+    this.activeRefresher.subscribe(() => {
+      this.refreshData();
+    });
+    this.refreshData();
+  }
+
+  public $onDestroy(): void {
+    this.activeRefresher.unsubscribe();
+
+  }
+}
+
+@DirectiveFactory('availabilityReaderService')
+class AvailabilityDirective implements ng.IDirective {
```

Comment:
Unless you're using `$compile` or `.$watch`, `.$digest`, or `.$apply` off of `$scope`, given our push to Typescript and ng2, I would prefer a component here instead and use `controllerAs` instead of `$scope`.